istioctl/cmd/analyze.go

```diff
@@ -16,21 +16,32 @@ package cmd
 
 import (
 	"fmt"
+	"io"
 	"os"
 	"strings"
 
 	"github.com/spf13/cobra"
 
 	"istio.io/istio/galley/pkg/config/analysis/analyzers"
+	"istio.io/istio/galley/pkg/config/analysis/diag"
 	"istio.io/istio/galley/pkg/config/analysis/local"
 	"istio.io/istio/galley/pkg/config/meta/metadata"
 	cfgKube "istio.io/istio/galley/pkg/config/source/kube"
 	"istio.io/istio/pkg/kube"
 )
 
+type FoundAnalyzeIssuesError struct {
+	int
+}
+
```

Comment:
Do we need this? 

I think we should decouple the reporting of validation messages (including how many) from the mechanism for setting an error code. The error code doesn't care how many messages there are. The user does, but we can output that as part of our normal reporting, in a number of ways, and tying it to the return code mechanism feels awkward to me.